plugins/ngrok/credentials.go

```diff
@@ -0,0 +1,102 @@
+package ngrok
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+	"gopkg.in/yaml.v3"
+)
+
+func Credentials() schema.CredentialType {
+	return schema.CredentialType{
+		Name:    credname.Credentials,
+		DocsURL: sdk.URL("https://ngrok.com/docs/ngrok-agent/config"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.AuthToken,
+				MarkdownDescription: "Auth Token used to authenticate to ngrok.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 43,
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.APIKey,
+				MarkdownDescription: "API Key used to authenticate to ngrok API.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 48,
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.TempFile(ngrokConfig, provision.Filename("config.yml"), provision.AddArgs("--config", "{{ .Path }}")),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			importer.MacOnly(
+				TryngrokConfigFile("~/Library/Application Support/ngrok/ngrok.yml"),
+			),
+			importer.LinuxOnly(
+				TryngrokConfigFile("~/.config/ngrok/ngrok.yml"),
+			),
+		)}
+}
+
+func ngrokConfig(in sdk.ProvisionInput) ([]byte, error) {
+	config := Config{
+		AuthToken: in.ItemFields[fieldname.AuthToken],
+		APIKey:    in.ItemFields[fieldname.APIKey],
+		Version:   "2", // required field for ngrok CLI to work
```

Comment:
Have a look at my other comment, I would like to use an environment variable provisioner to get rid of needing to hardcode this. If we can make a good argument for keeping it here, I would much rather import it from the config file, whenever possible, given it is one of the fields there:
https://ngrok.com/docs/ngrok-agent/config#global-options

Reply:
Makes sense to import from the config file instead of hardcoding. I’ll update today morning the first thing.

Reply:
We now import the ngrok version from the config file instead of hardcoding it: https://github.com/1Password/shell-plugins/pull/165/commits/09b80ad8b86c3bdea3d220b21ebfe30acf42a3f4.

Reply:
No problem about the back and forth. Always helps to evaluate all options for users' best experience. :) With this comment we should be good to go: https://github.com/1Password/shell-plugins/pull/165#issuecomment-1430525243.

Reply:
This will likely not be an issue anymore, if we decide to wait and go with envvars, which is my preferred approach 😄 

However, I do not agree that we should have this hard-coded. This does not take into consideration future possible version bumps in the config file format that `ngrok` may perform. 

That being said, we might reach the situation where the importer imports credentials from a config file with version "3", but provisions it to be "2", which is not guaranteed to work.

I don't think that we should be imposing more restrictions than the underlying CLI does, for its own configuration. In the case of this config's file version, this is something that the user has access to as well. Seems unnecessary to me to guard that, when the call to the 3rd party CLI is done through a plugin.

I'd love to hear what you both think!

Reply:
I guess one could argue that if the config file version is bumped, we should make a conscious choice to bump it plugin-side as well, to make sure what we provision is indeed compliant with the new format.

If this is indeed how we want to ration this, I think it would also be wise to only import from a config file if the format version is "2" as well, since otherwise the current parsing logic may not produce reliable results.

Reply:
Once again, I'd be in favour for waiting and using an env-var provisioner. But I think this idea could work as well, if we're really eager to push this out 😄 Thanks for the back and forth!

Reply:
> I guess one could argue that if the config file version is bumped, we should make a conscious choice to bump it plugin-side as well, to make sure what we provision is indeed compliant with the new format.

:100: this. The more complexity we can absorb in the plugin itself, the better experience we all have using the plugin. If something changes in the ngrok configfile format/version, only a single user of the plugin has to make the change in the plugin, and all of us will get the new and improved behavior. If we push setting/maitaining the version number to every single user by moving it into the 1Password item of all, everyone needs to maintain and update it separately in the case of a change.

If the way to pass a credential would differ between two major versions of a tool (e.g. ngrok), we can even add version detection in the plugin ecosystem, so that the behavior of the plugin is dependent on the version of the tool you have installed, so that it continues to work and you can switch between the versions seamlessly. We can tackle that implementation if and when needed, that should be relatively straightforward to add.

Reply:
Switching to an envvar provisioner is a non-breaking change, so we can always consider that in a next iteration of this plugin. The current implementation is already super valuable for users, so let's get it out there :rocket:

Reply:
Since the ngrok team already has a fix for API Token's envvar, we can push that update fairly soon. It's just a matter of few more days/weeks for the ngrok update to be available on the package managers they support. 🤞🏼